### PR TITLE
Show hardware information.

### PR DESCRIPTION
**Pull request by magalilemes**
This PR is far from ready, and it is a first attempt into showing hardware information from remote machines. To use it, run `kw statistics --remote IP PORT`, and then if everything is okay, it will show an output like the one below, providing information about the CPU, RAM, and type of storage devices.

```
CPU:
        Name: QEMU Virtual CPU version 2.5+
        Frequency:
                CPU MHz:             2304.066
RAM:
        Total RAM: 2037488 kB
Storage devices:
        fd0    rotational
        sda    rotational
        sr0    rotational
```

I haven't added tests yet, and the option still doesn't provide warnings when users mistakenly use the command. I also need to add another option for showing hardware information from the local machine. But before I proceed, I would like to receive feedback on what's been done so far.

### CODE REVIEW

**Comment by alan-barzilay** [comment]
have you tried inxi? it could be a quicker but dirtier way of accomplishing this.

**Comment by magalilemes** [comment]
@alan-barzilay I haven't. I didn't even know about this. I'll take a look. Thanks

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi Magali,

This is a very nice step toward the hardware information. I would like to suggest a different direction here:

Soon we will need a mechanism to capture hardware information from local, remote, and virtual machines. For this reason, we need something more robust here; I think you could create a new file named device_info_manager (please, see if you find a better name). This file will have functions for collecting this sort of data:
1. RAM
2. CPU
3. DISK
4. OS Device

This file will need some support function for getting those data from a local machine, remote, and virtual machine. Additionally, you will probably need something for dealing with a data store and formatting in that file. See the below draft:

```
# file: src/device_info_manager.sh
include 'remote.sh'

function get_ram()
{
  local target="$1"
  ...
  case "$target" in
    1) # VM_TARGET
    2) # LOCAL_TARGET
    3) # REMOTE
     ssh_to_remote_get_ram()
     ssh_to_remote_get_cpu()
     ...
     ;;
  esac
}

function get_cpu()
{
  # Same idea as above
}

function get_disk()
{
  # Same idea as above
}


function collect_hardware_data()
{

...
  mem=$(get_mem "param1" ...)
  cpu=$(get_mem "param1" ...)
...
  process_all_data()
  display_data()
}


```

Keep in mind that from the data statistics perspective, we probably want to separate data from the remote and local machines. Anyway, I see a series that looks like this:

1. Patch1: Basic data collect for local machine;
2. Patch2: Basic data collect for virtual machine;
3. Patch3: Basic data collect for remote machine;
4. Patch4: Find a way to associate host machines to statistics.

Finally, most of this information is natively available in all distro. For this reason, I don't recommend adding an external dependency (we don't need anything fancy).

File: src/statistics.sh [hunk 1]
```diff
@@ -19,34 +20,49 @@ declare -gA shared_data=( ["deploy"]='' ["build"]='' ["list"]='' ["uninstall"]='
...
-  local date_param
 
   if [[ "$1" == -h ]]; then
     statistics_help
     exit 0
   fi
 
-  shift 1 # Remove the first option, i.e., --day, --week, or --year
-  local date_param="$@"
+  if [[ ! $info_request == 'remote' ]]; then
```
[the remark this answers is not in the document]
Comment (magalilemes) [reply src/statistics.sh:29]: Ok! I'll change `info_request` to `target` when refactoring statistics in #276. wdyt?

[the remark this answers is not in the document]
Comment (rodrigosiqueira) [reply src/statistics.sh:29]: Btw, I think you need to focus on my comments in the conversation section. I think you'll need to rethink your approach here.

[the remark this answers is not in the document]
Comment (magalilemes) [reply src/statistics.sh:29]: Yes, I'm focusing on it, and I like all this new direction you pointed out in the conversation better.

[the remark this answers is not in the document]
Comment (rodrigosiqueira) [reply src/statistics.sh:29]: Nice. I think the new set of changes will be much different from this one; for this reason, I would like to suggest you close this PR and open a new one when you have another version.

**Pull request closed without merging**